--- domainauth.py.org	2019-02-25 12:00:37.000000000 +0100
+++ domainauth.py	2019-02-25 12:02:12.939509640 +0100
@@ -294,9 +294,9 @@
 
         blength = self.config.getboolean(self.section, 'signbodylength')
 
-        dkimhdr = sign(message, selector, domain, privkeycontent, canonicalize=canon,
+        dkimhdr = sign(message, selector, str.encode(domain), str.encode(privkeycontent), canonicalize=canon,
                        include_headers=inc_headers, length=blength, logger=suspect.get_tag('debugfile'))
-        if dkimhdr.startswith('DKIM-Signature: '):
+        if dkimhdr.decode().startswith('DKIM-Signature: '):
             dkimhdr = dkimhdr[16:]
 
         suspect.addheader('DKIM-Signature', dkimhdr, immediate=True)
